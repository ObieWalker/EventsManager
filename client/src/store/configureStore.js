import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;