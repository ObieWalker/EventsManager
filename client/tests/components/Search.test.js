import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connectSearch, { Search } from '../../src/components/Search';

let mountedSearch;
let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  }
};

const loginUser = { isAuthenticated: false, user: {} };
const getAllCenters = jest.fn(() => Promise.resolve());
const clearCenterState = jest.fn(() => Promise.resolve());
const changeSearchState = jest.fn(() => Promise.resolve());
const resetPage = jest.fn(() => Promise.resolve());
const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedSearch) {
    props = {
      getAllCenters,
      clearCenterState,
      changeSearchState,
      resetPage
    };
    history.push = jest.fn();
    mountedSearch = shallow(<Search {...props} />);
  }
  return mountedSearch;
};

describe('Login Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('renders component successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updates state on input change', () => {
    wrapper.find('#filter').simulate('change', {
      target: { name: 'filter', value: 'Lekki' }
    });
    expect(wrapper.state().filter).toBe('Lekki');
  });

  it('updates state on input change', () => {
    wrapper.find('#capacity').simulate('change', {
      target: { name: 'capacity', value: 500 }
    });
    expect(wrapper.state().capacity).toBe(500);
  });

  it('updates state on input change', () => {
    wrapper.find('#facility').simulate('change', {
      target: { name: 'facility', value: 'chairs' }
    });
    expect(wrapper.state().facility).toBe('chairs');
  });

  it('testing Search function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().handleSearch(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });
  it('testing fetch all centers function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().fetchCenters(e);
    expect(e.preventDefault).toHaveBeenCalled();
    expect(wrapper.state().filter).toBe('');
    expect(wrapper.state().capacity).toBe('');
    expect(wrapper.state().facility).toBe('');
  });
});
describe('Connected Component', () => {
  describe('Connected mountedSearch', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        loginUser
      });
      wrapper = shallow(<connectSearch store={store} history={history} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
