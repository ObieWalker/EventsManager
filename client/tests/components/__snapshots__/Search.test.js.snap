// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    changeSearchState={[MockFunction]}
    clearCenterState={[MockFunction]}
    getAllCenters={[MockFunction]}
    resetPage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="col s12 full-width"
      >
        <div
          className="input-field col s12 l3"
        >
          <label
            htmlFor="filter"
          >
            Name or Location
          </label>
          <input
            className="validate"
            id="filter"
            name="filter"
            onChange={[Function]}
            type="search"
            value=""
          />
        </div>
        <div
          className="input-field col s12 l4"
        >
          <label
            htmlFor="facility"
          >
            Facility
          </label>
          <input
            className="validate"
            id="facility"
            name="facility"
            onChange={[Function]}
            type="search"
            value=""
          />
        </div>
        <div
          className="input-field col s12 l1"
        >
          <label
            htmlFor="capacity"
          >
            Capacity
          </label>
          <input
            className="validate"
            id="capacity"
            min="1"
            name="capacity"
            onChange={[Function]}
            type="number"
            value=""
          />
        </div>
        <div
          className="input-field col s12 l4"
        >
          <button
            className="btn peach-gradient btn-sm"
            onClick={[Function]}
            type="submit"
          >
            search
          </button>
          <button
            className="btn peach-gradient btn-sm"
            onClick={[Function]}
          >
            Clear Search
          </button>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-field col s12 l3"
          >
            <label
              htmlFor="filter"
            >
              Name or Location
            </label>
            <input
              className="validate"
              id="filter"
              name="filter"
              onChange={[Function]}
              type="search"
              value=""
            />
          </div>,
          <div
            className="input-field col s12 l4"
          >
            <label
              htmlFor="facility"
            >
              Facility
            </label>
            <input
              className="validate"
              id="facility"
              name="facility"
              onChange={[Function]}
              type="search"
              value=""
            />
          </div>,
          <div
            className="input-field col s12 l1"
          >
            <label
              htmlFor="capacity"
            >
              Capacity
            </label>
            <input
              className="validate"
              id="capacity"
              min="1"
              name="capacity"
              onChange={[Function]}
              type="number"
              value=""
            />
          </div>,
          <div
            className="input-field col s12 l4"
          >
            <button
              className="btn peach-gradient btn-sm"
              onClick={[Function]}
              type="submit"
            >
              search
            </button>
            <button
              className="btn peach-gradient btn-sm"
              onClick={[Function]}
            >
              Clear Search
            </button>
          </div>,
        ],
        "className": "col s12 full-width",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="filter"
              >
                Name or Location
              </label>,
              <input
                className="validate"
                id="filter"
                name="filter"
                onChange={[Function]}
                type="search"
                value=""
              />,
            ],
            "className": "input-field col s12 l3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name or Location",
                "htmlFor": "filter",
              },
              "ref": null,
              "rendered": "Name or Location",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "filter",
                "name": "filter",
                "onChange": [Function],
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="facility"
              >
                Facility
              </label>,
              <input
                className="validate"
                id="facility"
                name="facility"
                onChange={[Function]}
                type="search"
                value=""
              />,
            ],
            "className": "input-field col s12 l4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Facility",
                "htmlFor": "facility",
              },
              "ref": null,
              "rendered": "Facility",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "facility",
                "name": "facility",
                "onChange": [Function],
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="capacity"
              >
                Capacity
              </label>,
              <input
                className="validate"
                id="capacity"
                min="1"
                name="capacity"
                onChange={[Function]}
                type="number"
                value=""
              />,
            ],
            "className": "input-field col s12 l1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Capacity",
                "htmlFor": "capacity",
              },
              "ref": null,
              "rendered": "Capacity",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "capacity",
                "min": "1",
                "name": "capacity",
                "onChange": [Function],
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn peach-gradient btn-sm"
                onClick={[Function]}
                type="submit"
              >
                search
              </button>,
              <button
                className="btn peach-gradient btn-sm"
                onClick={[Function]}
              >
                Clear Search
              </button>,
            ],
            "className": "input-field col s12 l4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "search",
                "className": "btn peach-gradient btn-sm",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "search",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Clear Search",
                "className": "btn peach-gradient btn-sm",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Clear Search",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="col s12 full-width"
        >
          <div
            className="input-field col s12 l3"
          >
            <label
              htmlFor="filter"
            >
              Name or Location
            </label>
            <input
              className="validate"
              id="filter"
              name="filter"
              onChange={[Function]}
              type="search"
              value=""
            />
          </div>
          <div
            className="input-field col s12 l4"
          >
            <label
              htmlFor="facility"
            >
              Facility
            </label>
            <input
              className="validate"
              id="facility"
              name="facility"
              onChange={[Function]}
              type="search"
              value=""
            />
          </div>
          <div
            className="input-field col s12 l1"
          >
            <label
              htmlFor="capacity"
            >
              Capacity
            </label>
            <input
              className="validate"
              id="capacity"
              min="1"
              name="capacity"
              onChange={[Function]}
              type="number"
              value=""
            />
          </div>
          <div
            className="input-field col s12 l4"
          >
            <button
              className="btn peach-gradient btn-sm"
              onClick={[Function]}
              type="submit"
            >
              search
            </button>
            <button
              className="btn peach-gradient btn-sm"
              onClick={[Function]}
            >
              Clear Search
            </button>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-field col s12 l3"
            >
              <label
                htmlFor="filter"
              >
                Name or Location
              </label>
              <input
                className="validate"
                id="filter"
                name="filter"
                onChange={[Function]}
                type="search"
                value=""
              />
            </div>,
            <div
              className="input-field col s12 l4"
            >
              <label
                htmlFor="facility"
              >
                Facility
              </label>
              <input
                className="validate"
                id="facility"
                name="facility"
                onChange={[Function]}
                type="search"
                value=""
              />
            </div>,
            <div
              className="input-field col s12 l1"
            >
              <label
                htmlFor="capacity"
              >
                Capacity
              </label>
              <input
                className="validate"
                id="capacity"
                min="1"
                name="capacity"
                onChange={[Function]}
                type="number"
                value=""
              />
            </div>,
            <div
              className="input-field col s12 l4"
            >
              <button
                className="btn peach-gradient btn-sm"
                onClick={[Function]}
                type="submit"
              >
                search
              </button>
              <button
                className="btn peach-gradient btn-sm"
                onClick={[Function]}
              >
                Clear Search
              </button>
            </div>,
          ],
          "className": "col s12 full-width",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="filter"
                >
                  Name or Location
                </label>,
                <input
                  className="validate"
                  id="filter"
                  name="filter"
                  onChange={[Function]}
                  type="search"
                  value=""
                />,
              ],
              "className": "input-field col s12 l3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name or Location",
                  "htmlFor": "filter",
                },
                "ref": null,
                "rendered": "Name or Location",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "validate",
                  "id": "filter",
                  "name": "filter",
                  "onChange": [Function],
                  "type": "search",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="facility"
                >
                  Facility
                </label>,
                <input
                  className="validate"
                  id="facility"
                  name="facility"
                  onChange={[Function]}
                  type="search"
                  value=""
                />,
              ],
              "className": "input-field col s12 l4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Facility",
                  "htmlFor": "facility",
                },
                "ref": null,
                "rendered": "Facility",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "validate",
                  "id": "facility",
                  "name": "facility",
                  "onChange": [Function],
                  "type": "search",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="capacity"
                >
                  Capacity
                </label>,
                <input
                  className="validate"
                  id="capacity"
                  min="1"
                  name="capacity"
                  onChange={[Function]}
                  type="number"
                  value=""
                />,
              ],
              "className": "input-field col s12 l1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Capacity",
                  "htmlFor": "capacity",
                },
                "ref": null,
                "rendered": "Capacity",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "validate",
                  "id": "capacity",
                  "min": "1",
                  "name": "capacity",
                  "onChange": [Function],
                  "type": "number",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn peach-gradient btn-sm"
                  onClick={[Function]}
                  type="submit"
                >
                  search
                </button>,
                <button
                  className="btn peach-gradient btn-sm"
                  onClick={[Function]}
                >
                  Clear Search
                </button>,
              ],
              "className": "input-field col s12 l4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "search",
                  "className": "btn peach-gradient btn-sm",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "search",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Clear Search",
                  "className": "btn peach-gradient btn-sm",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Clear Search",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
